### ISSUE TITLE
add option to save observed streamflow data used for calculating performance metrics

### ISSUE THREAD

**Issue by Ben-Choat** [NOAA-OWP/ngen-cal#93]
Short description explaining the high-level reason for the new issue.

ENHANCEMENT

## Current behavior
The ngen-cal cod euses HY_Features to access nwis data to get usgs streamflow data. It uses this data to calculate performance metrics against the simulated streamflow. The data is not saved though.

## Desired behavior
Include options in the calibration configuration file to 1. Specifiy if the user would like to save the observed data and 2. if 1 is True, provide a directory in which to save the data.

**Comment by aaraney** [comment]
@hellkite500, I think we can accomplish this and support other use cases if we create something like an `ngen_cal_model_evaluate` hook. A plugin that returns `None` could be used in this case just to read the values that are passed to evaluate. Its not perfect, but I think we should consider it.

**Comment by aaraney** [comment]
@Ben-Choat, once #155 is merged you will be able to add this functionality via `ngen.cal`'s plugin system. See #155 for an example.
